### ISSUE TITLE
Propfind should deliver StorageId property

### ISSUE THREAD

**Issue by JanAckermann** [owncloud/ocis#3345]
## Is your feature request related to a problem? Please describe.
For various reasons, we need the storageId, which is assigned to a File in the WebUI.

Currently, it might occur in the FileId, which consists of storageId:fileId.
This is **not** the case if you do a propfind on the trash bin or spaces trash bin.

Therefore it would be good to have the possibility to retrieve the storageId on all propfinds via a dedicated property. 

## Describe the solution you'd like
Add a storageID DAV property

**Comment by C0rby** [comment]
@butonic, do you see any problems with introducing an extra property for the storage id?
Except that it will be oCIS exclusive.

If not I'll just implement it.

**Comment by butonic** [comment]
Yes, please! I'd prefer `spaceid` and `fileid`. And having that property in oc10 would be great as well.

**Comment by butonic** [comment]
hmmm. if you don't get a `spaceid` in trashbin responses that is a bug.

**Comment by butonic** [comment]
In the bigger picture we could take the oc10 numeric storage id as the storageid. But that would create problems when trying to migrate spaces to other storage providers. One solution would be to prefix the numeric storage id with the instance id (if we want to leak that), but that only makes collisions less likely. While `spaceid`s in the form of `{instanceid}-{numericstosrageid}` might work, I'd prefer a real uuid.

Now regarding a seperate id: makes sense, however the graph api will always take a single `id` property as the identifier for a resource. It does contain delimiters: `!` and `$`. In `C12644A14B0A7750!6180`, `C12644A14B0A7750` is my userid as well as my `personal` drive id. `6180` is an autoincrement id for resources in my drive. At least that is how MS implemented it in onedrive.

To clients the fileid is an opaque string. It is made of `{spaceid}!{nodeid}` so we can efficiently route requests to the rosponsible storage space provider. Clients must not parse it, because the server might change the way the opaqueid is constructed at any time, eg introduce more prefixes for further sharding the spaces.

To let clients link resources to spaces we should add an `oc:spaceid` property that contains just the `spaceid`.

**Comment by C0rby** [comment]
Ok, I'll implement this next week. :+1:

**Comment by C0rby** [comment]
This is now included in the current oCIS master.
The new property is called `<oc:spaceid/>`.
